### DISCUSSION TITLE
Reported Value vs Recognized Value Different

### DISCUSSION THREAD

**Discussion by lowson** [jomjol/AI-on-the-edge-device#3201] [General] [answered]
### The Problem

So I have noticed two issues for awhile and I can't tell what is causing them or how to fix. I have tweaked various settings and models and update the to latest version, its not "critical" but looking for some help to debug what is happening.

First issue is the Recognized Digit ROIs reporting 0 as 10

Example
![Screenshot from 2024-01-26 18-35-06](https://github.com/jomjol/AI-on-the-edge-device/assets/5473528/9c9c5096-df2f-4aba-ab90-9cf4eb271182)

This is reporting the value 2419 right now and typically won't update until it sees 2421 likely.

Second issue is the Recognized Digit ROIs reporting correctly but not update the posted Value

>> sorry don't have a photo but ill update with one, 

Will be in a state similar to the above but the Recognized Digit ROIs are 100% correct but the Value update doesn't change, in the above it would be stuck at 2419 but Recognized 2420 and again won't update until another full digit change.

I am trying to monitor if its specific digits or not but I haven't noticed anything specific yet, since the issue does resolve after another cycle I haven't report it until now. 

### Version

Release: v15.4.0 (Commit: 74d4f20)

### Logfile

```shell
No Errors :/
```

**Comment by lowson** [comment]
![Screenshot from 2024-01-26 19-25-22](https://github.com/jomjol/AI-on-the-edge-device/assets/5473528/eef5c881-f2b5-421c-a84d-2b98aab1d71a)

Flipped once the reading changed to 0.1 vs 10.

Anyway to force the digit to only real to a max of 9 for the first example

**Comment by Slider0007** [comment]
On recognition page single ROI values are getting rounded (e.g. 9.95 -> 10.0), internally for processing the algo they are truncated (e.g. 9.95 -> 9), though. This means value only gets detected as 0 if value is for sure greater than 0. Therefore it can happen displayed ROI values seems not matching in some situations...

**Comment by spanzetta** [comment]
> On recognition page single ROI values are getting rounded (e.g. 9.95 -> 10.0), internally for processing the algo they are truncated (e.g. 9.95 -> 9), though. This means value only gets detected as 0 if value is for sure greater than 0. Therefore it can happen displayed ROI values seems not matching in some situations...

Could it be that sometimes the internal processing get rounded instead of get truncated?

**Comment by spanzetta** [comment]
> > On recognition page single ROI values are getting rounded (e.g. 9.95 -> 10.0), internally for processing the algo they are truncated (e.g. 9.95 -> 9), though. This means value only gets detected as 0 if value is for sure greater than 0. Therefore it can happen displayed ROI values seems not matching in some situations...
> 
> Could it be that sometimes the internal processing get rounded instead of get truncated?

With latest 15.5 it is happening many times..

**Comment by spanzetta** [comment]
See here.. it should be 8 and not 9

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/c1f8db23-8be5-4028-9a55-dfc919cfa055)

**Comment by simowilso** [comment]
![Screenshot_20240207-094947](https://github.com/jomjol/AI-on-the-edge-device/assets/15340247/d1cc3ed1-d182-4fe3-b0db-50a8a1f73bc8)
I'm seeing the same thing

**Comment by spanzetta** [comment]
This is even more strange

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/29dd2ae0-cec3-444d-ab9a-d247b036a164)

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/ab8cae24-281f-4f6f-a14b-5be9f1be4171)

**Comment by Slider0007** [comment]
> This is even more strange
> 
> ![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/29dd2ae0-cec3-444d-ab9a-d247b036a164)
> 
> ![image](https://github.com/jomjol/AI-on-the-edge-device/assets/49818743/ab8cae24-281f-4f6f-a14b-5be9f1be4171)
> 

I assume this issue here is also related to change introduced with #2778. Some further discussions, see #2857

@rainman110: FYI

**Comment by caco3** [comment]
We reverted https://github.com/jomjol/AI-on-the-edge-device/pull/2778 for now. Please try rolling for a version as it was before https://github.com/jomjol/AI-on-the-edge-device/pull/2778.

**Comment by petercoxphoto** [comment]
I'm having the same issue here. Build 15.7.0. 

![IMG_4293](https://github.com/user-attachments/assets/fe8ce3ae-7388-43cc-90e4-ded825596550)

**Comment by friedpa** [comment]
Your picture has (almost) no contrast and sharpness. 
Example where you should end up....

![grafik](https://github.com/user-attachments/assets/1c395d63-4a97-46e5-9529-4d9431e2d27d)

**Comment by petercoxphoto** [comment]
Granted. But the system still recognizes the digits correctly in the detection screen. So there's sufficient contrast to do that reliably. 

The question is why is the reported value different from what the detection shows?

**Comment by Vint1234** [comment]
I am having a similar issue with my new LCD based water meter on Build 15.7.0. The recognition shows the correct digits as displayed on the meter, but the value that is finally transmitted has an 8 instead of a 7, resulting in 2841 instead of 2741, see pictures below.

I am using the dig_class11_1700 mode, as it is the most appropriate for a simple LCD without continuos rotation of digits.

![grafik](https://github.com/user-attachments/assets/aef99251-24c0-4e6e-a547-19170b278101)

![grafik](https://github.com/user-attachments/assets/c3068c97-d8db-48b7-b254-45567ee26011)

**Comment by Vint1234** [comment]
Just did a quick test, setting Check Digit Increase Consistency to false, as I do not see the point in a display with "perfect" digit transistions. First round emmediately reported the now expected negative rate error, as 2741 was recognised but 2841 was last sent. I will see, if this fixes it, once I have consumed more water.

**Comment by caco3** [comment]
@Vint1234 Since the digits are italics, it can help to rotate the image a bit to work against the tilting of the digits.

**Comment by Vint1234** [comment, marked as answer]
I have tested this now for 5 days and did not have any issues with, since I set Check Digit Increase Consistency to false. The italic digits do not seem to pose any issue with the recognition. My meter is a Diehl Hydrus2. The LED turns on 20s before the picture is taken and the LED is positioned in such a way, that it is above the light sensor of the meter that turns the display on.
